test/IISIntegration.IISServerFunctionalTests/Utilities/IISExpressSupportsInProcessHostingAttribute.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Testing.xunit;
+
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests
+{
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class IISExpressSupportsInProcessHostingAttribute : Attribute, ITestCondition
+    {
+        public bool IsMet => AncmSchema.SupportsInProcessHosting;
+
+        public string SkipReason => AncmSchema.SkipReason;
+
+        private class AncmSchema
+        {
+            public static bool SupportsInProcessHosting { get; }
+            public static string SkipReason { get; } = "IIS Express must be upgraded to support in-process hosting.";
+
+            static AncmSchema()
+            {
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    SkipReason = "IIS Express tests can only be run on Windows";
+                    return;
+                }
+
+                var ancmConfigPath = Path.Combine(
```

Comment:
Nit: we may want to check x64/x86 based on bitness of the process. Even though the files are the same, the bitness will correspond to the config that is loaded, ex x64 will use ProgramFiles/... and x86 will use ProgramFiles/x86....

Reply:
I'll just use Environment.SpecialFolder.ProgramFiles. This defaults to the programfiles that matches the bitness of the PC.